### PR TITLE
feat: Bearing rating to linear one

### PR DESCRIPTION
**Pull request by BonyBabu**

### CODE REVIEW

File: map/src/test/java/openlr/map/utils/GeometryUtilsTest.java [hunk 1]
```diff
@@ -480,6 +491,12 @@ public final void testCoordinateBounds() {
                 Double.MIN_VALUE));
     }
 
+    @Test
+    public void bearingDifferenceTest() {
```
Comment (stephencurran) [inline map/src/test/java/openlr/map/utils/GeometryUtilsTest.java]: A couple of more tests where the first angle isn't 0 would be good.  eg.

```java
assertEquals(GeometryUtils.bearingDifference(345, 10), 25);
assertEquals(GeometryUtils.bearingDifference(150, 220), 70);
```

File: map/src/main/java/openlr/map/utils/GeometryUtils.java [hunk 2]
```diff
@@ -702,6 +716,21 @@ public static GeoCoordinates determineCoordinateInDistance(
         return new GeoCoordinatesImpl(lon2, lat2);
     }
 
+    /**
+     * Degree of interval between two angles
+     * @param firstBearing angle of line to the north
+     * @param secondBearing angle of line to the north
+     * @return Absolute rounded interval value
```
Comment (stephencurran) [inline map/src/main/java/openlr/map/utils/GeometryUtils.java]: I don't think the return result has the correct description.  Maybe "the absolute difference between the 2 angles".

File: map/src/main/java/openlr/map/utils/GeometryUtils.java [hunk 3]
```diff
@@ -702,6 +716,21 @@ public static GeoCoordinates determineCoordinateInDistance(
         return new GeoCoordinatesImpl(lon2, lat2);
     }
 
+    /**
+     * Degree of interval between two angles
+     * @param firstBearing angle of line to the north
```
Comment (stephencurran) [inline map/src/main/java/openlr/map/utils/GeometryUtils.java]: "angle of line to the north" makes it sound like there is a line going to the north.  Maybe "angle of a line relative to due north".

File: decoder/src/main/java/openlr/decoder/properties/OpenLRDecoderProperty.java [hunk 4]
```diff
@@ -37,6 +37,28 @@
  * <p>
  * Address: TomTom International B.V., Oosterdoksstraat 114, 1011DK Amsterdam,
  * the Netherlands
+ * <p>
```
Comment (stephencurran) [inline decoder/src/main/java/openlr/decoder/properties/OpenLRDecoderProperty.java]: Let's see what is adding these headers and disable it.

File: map/src/test/java/openlr/map/utils/GeometryUtilsTest.java (hunk 1 above)
Comment (BonyBabu) [reply map/src/test/java/openlr/map/utils/GeometryUtilsTest.java]: 458ee63d9b353f59b5145a2bb129eef2fe2ab6ad

File: map/src/main/java/openlr/map/utils/GeometryUtils.java (hunk 2 above)
Comment (BonyBabu) [reply map/src/main/java/openlr/map/utils/GeometryUtils.java]: 458ee63d9b353f59b5145a2bb129eef2fe2ab6ad

File: map/src/main/java/openlr/map/utils/GeometryUtils.java (hunk 3 above)
Comment (BonyBabu) [reply map/src/main/java/openlr/map/utils/GeometryUtils.java]: 458ee63d9b353f59b5145a2bb129eef2fe2ab6ad

File: map/src/test/java/openlr/map/utils/GeometryUtilsTest.java (hunk 1 above)
Comment (BonyBabu) [reply map/src/test/java/openlr/map/utils/GeometryUtilsTest.java]: 78af58f0f964ce6e2c59d398baa21759cfa62a88

**Pull request merged**